lifecycle/README.rst

```diff
@@ -0,0 +1,353 @@
+          :alt: asciicast
+
+
+Alternatively, these three programs can be run together in the same terminal using the launch file (as of ROS 2 Bouncy):
+
+.. code-block:: bash
+
+   ros2 launch lifecycle lifecycle_demo.launch.py
+
+If we look at the output of the ``lifecycle_talker``\ , we notice that nothing seems to happen.
+And this does make sense, since every node starts as ``unconfigured``.
+The lifecycle_talker is not configured yet and in our example, no publishers and timers are created yet.
+The same behavior can be seen for the ``lifecycle_listener``\ , which is less surprising given that no publishers are available at this moment.
+The interesting part starts with the third terminal.
+In there we launch our ``lifecycle_service_client`` which is responsible for changing the states of the ``lifecycle_talker``.
+
+Triggering transition 1 (configure)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+   [lc_client] Transition 1 successfully triggered.
+   [lc_client] Node lc_talker has current state inactive.
+
+makes the lifecycle talker change its state to inactive.
+Inactive means that all publishers and timers are created and configured.
+However, the node is still not active.
+Therefore no messages are getting published.
+
+.. code-block:: bash
+
+   [lc_talker] on_configure() is called.
+   Lifecycle publisher is currently inactive. Messages are not published.
+   ...
+
+The lifecycle listener on the same time receives a notification as it listens to every state change notification of the lifecycle talker.
+In fact, the listener receives two consecutive notifications.
+One for changing from the primary state "unconfigured" to "configuring".
+Because the configuring step was successful within the lifecycle talker, a second notification from "configuring" to "inactive".
+
+.. code-block:: bash
+
+   [lc_listener] notify callback: Transition from state unconfigured to configuring
+   [lc_listener] notify callback: Transition from state configuring to inactive
+
+Triggering transition 2 (activate)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+   [lc_client] Transition 2 successfully triggered.
+   [lc_client] Node lc_talker has current state active.
+
+makes the lifecycle talker change its state to active.
+Active means that all publishers and timers are now activated.
+Therefore the messages are now getting published.
+
+.. code-block:: bash
+
+   [lc_talker] on_activate() is called.
+   [lc_talker] Lifecycle publisher is active. Publishing: [Lifecycle HelloWorld #11]
+   [lc_talker] Lifecycle publisher is active. Publishing: [Lifecycle HelloWorld #12]
+   ...
+
+The lifecycle listener receives the same set of notifications as before.
+Lifecycle talker changed its state from inactive to active.
+
+.. code-block:: bash
+
+   [lc_listener] notify callback: Transition from state unconfigured to configuring
+   [lc_listener] notify callback: Transition from state configuring to inactive
```

Comment:
I see two more transitions after these before the listener receives data

```
[INFO] [lc_listener]: notify callback: Transition from state inactive to activating
[INFO] [lc_listener]: notify callback: Transition from state activating to active
```

Reply:
good catch. That was a copy paste error.